codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/flexiblechecksums/FlexibleChecksumsRequestIntegration.kt

```diff
@@ -0,0 +1,68 @@
+package software.amazon.smithy.aws.swift.codegen.customization.flexiblechecksums
+
+import software.amazon.smithy.aws.traits.HttpChecksumTrait
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.getOrNull
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.integration.middlewares.handlers.MiddlewareShapeUtils
+import software.amazon.smithy.swift.codegen.middleware.MiddlewarePosition
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderable
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareStep
+import software.amazon.smithy.swift.codegen.middleware.OperationMiddleware
+import software.amazon.smithy.swift.codegen.model.expectShape
+import software.amazon.smithy.swift.codegen.model.shapes
+
+class FlexibleChecksumsRequestIntegration : SwiftIntegration {
+    override fun enabledForService(model: Model, settings: SwiftSettings): Boolean = model
+        .shapes<OperationShape>()
+        .any { it.hasTrait(HttpChecksumTrait::class.java) }
+
+    override fun customizeMiddleware(
+        ctx: ProtocolGenerator.GenerationContext,
+        operationShape: OperationShape,
+        operationMiddleware: OperationMiddleware,
+    ) {
+        val httpChecksumTrait = operationShape.getTrait(HttpChecksumTrait::class.java).getOrNull()
```

Comment:
Is this the same `.getOrNull()` that I'm eliminating in the XML branch?

Reply:
(I think we decided to use `.orElse(null)` which is a Kotlin built-in)

Reply:
the `.orElse(null)` we were talking about I believe works on Java types not Kotlin types where we prefer to use `?.` that work on optionals. I can check where .getOrNull() is used and see if we can switch it over to the kotlin optionals